app/scripts/lib/multichain-api/caip-permission-adapter-middleware.test.js

```diff
@@ -0,0 +1,175 @@
+    end,
+    getCaveat,
+    getNetworkConfigurationByNetworkClientId,
+    handler,
+  };
+};
+
+describe('CaipPermissionAdapterMiddleware', () => {
+  describe('BARAD_DUR feature flag is not set', () => {
+    beforeAll(() => {
+      delete process.env.BARAD_DUR;
+    });
+
+    it('allows the request when BARAD_DUR feature flag is not set', async () => {
+      const { handler, next } = createMockedHandler();
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('does not read the permission state', async () => {
+      const { handler, getCaveat } = createMockedHandler();
+      await handler(baseRequest);
+      expect(getCaveat).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('BARAD_DUR feature flag is set', () => {
+    beforeAll(() => {
+      process.env.BARAD_DUR = 1;
+    });
+
+    it('gets the authorized scopes from the CAIP-25 endowement permission', async () => {
+      const { handler, getCaveat } = createMockedHandler();
+      await handler(baseRequest);
+      expect(getCaveat).toHaveBeenCalledWith(
+        'http://test.com',
+        Caip25EndowmentPermissionName,
+        Caip25CaveatType,
+      );
+    });
+
+    it('allows the request when there is no CAIP-25 endowement permission', async () => {
+      const { handler, getCaveat, next } = createMockedHandler();
+      getCaveat.mockReturnValue(null);
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('allows the request when the CAIP-25 endowement permission was not granted from the multichain flow', async () => {
+      const { handler, getCaveat, next } = createMockedHandler();
+      getCaveat.mockReturnValue({
+        value: {
+          isMultichainOrigin: false,
+        },
+      });
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('gets the chainId for the request networkClientId', async () => {
+      const { handler, getNetworkConfigurationByNetworkClientId } =
+        createMockedHandler();
+      await handler(baseRequest);
+      expect(getNetworkConfigurationByNetworkClientId).toHaveBeenCalledWith(
+        'mainnet',
+      );
+    });
+
+    it('throws an error if the requested scope method is not authorized in either the current scope or the wallet scope', async () => {
+      const { handler, end } = createMockedHandler();
+
+      await handler({
+        ...baseRequest,
+        method: 'unauthorized_method',
+      });
+      expect(end).toHaveBeenCalledWith(providerErrors.unauthorized());
+    });
+
+    it('allows the request if the requested scope method is authorized in the current scope', async () => {
+      const { handler, next } = createMockedHandler();
+
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('allows the request if the requested scope method is authorized in the wallet scope and the current scope does exist in the authorization', async () => {
```

Comment:
"the current scope" is a bit ambiguous. Not sure what it means in this context

Reply:
```suggestion
    it('allows the request if the requested method is authorized in the wallet scope and the scope specified by the request is authorized with at least one method or notification', async () => {
```

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/26296/commits/a3048931c6116c4fcda640da5cb34457f44f4bb5